ScanCommand/src/main/java/io/ballerina/scan/ScanCmd.java

```diff
@@ -0,0 +1,133 @@
+    private String targetDir;
+
+    @CommandLine.Option(names = {"--scan-report"}, description = "Enable scan report generation")
+    private boolean scanReport;
+
+    @CommandLine.Option(names = {"--list-rules"},
+            description = "List the internal rules available in the Ballerina scan tool.")
+    private boolean listRules;
+
+    @CommandLine.Option(names = {"--include-rules"},
+            converter = StringToListConverter.class,
+            description = "Specify the comma separated list of rules to include specific analysis issues")
+    private List<String> includeRules = new ArrayList<>();
+
+    @CommandLine.Option(names = {"--exclude-rules"},
+            converter = StringToListConverter.class,
+            description = "Specify the comma separated list of rules to exclude specific analysis issues")
+    private List<String> excludeRules = new ArrayList<>();
+
+    @CommandLine.Option(names = {"--platforms"},
+            converter = StringToListConverter.class,
+            description = "static code analysis output platform",
+            defaultValue = "local")
+    private List<String> platforms = new ArrayList<>();
+
+    public ScanCmd() {
+        this.outputStream = System.out;
+    }
+
+    @Override
+    public String getName() {
+        return "scan";
+    }
+
+    @Override
+    public void printLongDesc(StringBuilder out) {
+        StringBuilder builder = helpMessage();
+        out.append(builder);
+    }
+
+    @Override
+    public void printUsage(StringBuilder out) {
+        out.append("Tool for providing static code analysis results for Ballerina projects");
+    }
+
+    public StringBuilder helpMessage() {
+        InputStream inputStream = ScanCmd.class.getResourceAsStream("/cli-help/ballerina-scan.help");
+        StringBuilder builder = new StringBuilder();
+        if (inputStream != null) {
+            try (
+                    InputStreamReader inputStreamReader = new InputStreamReader(inputStream, StandardCharsets.UTF_8);
+                    BufferedReader br = new BufferedReader(inputStreamReader)
+            ) {
+                String content = br.readLine();
+                builder.append(content);
+                while ((content = br.readLine()) != null) {
+                    builder.append("\n").append(content);
+                }
+            } catch (IOException e) {
+                builder.append("Helper text is not available.");
+                throw new RuntimeException(e);
+            }
+        }
+
+        return builder;
+    }
+
+    @Override
+    public void setParentCmdParser(CommandLine parentCmdParser) {
+
+    }
+
+    // Scan Command Main Method
+    @Override
+    public void execute() {
+        if (helpFlag) {
+            StringBuilder builder = helpMessage();
+            outputStream.println(builder);
+            return;
+        }
+
+        outputStream.println();
+        outputStream.println("Running Scans...");
```

Comment:
Trying to understand. Why don't we have any tool execution logic in the `execute` method?

Reply:
The tool execution logic is to be put from the next PR onwards